### PR TITLE
Use SeedTemplate instead of Seed in GardenletConfiguration

### LINKED ISSUE #3469: Using SeedTemplate instead of Seed in GardenletConfiguration
**How to categorize this issue?**

/area control-plane
/kind enhancement
/priority normal

**What would you like to be added**:
The design of the `GardenletConfiguration` resource should be improved:
* Currently, it embeds `Seed` in `SeedConfig`: https://github.com/gardener/gardener/blob/a1b0891be6061c9511b7d6038e381c4a1a6c0f04/pkg/gardenlet/apis/config/types.go#L288-L291. Instead, it should embed `SeedTemplate`: https://github.com/gardener/gardener/blob/a1b0891be6061c9511b7d6038e381c4a1a6c0f04/pkg/apis/seedmanagement/v1alpha1/types_managedseed.go#L75-L83 using the internal version in `config` and the external one in `v1alpha1`. 
* Currently, it does not validate the `Seed` embedded via `SeedConfig`. The `SeedTemplate` should be validated by calling a function similar to e.g. https://github.com/gardener/gardener/blob/a1b0891be6061c9511b7d6038e381c4a1a6c0f04/pkg/apis/seedmanagement/validation/managedseed.go#L114-L128. 
* The explicit validation of `SeedConfig` should be removed from `seedmanagement/validation`.
* Potentially, `SeedTemplate` should be also moved to `core` (and its validation to `core/validation`) to avoid the extra dependency to `seedmanagement`. This dependency would not lead to a circular dependency because `GardenletConfiguration` is not used explicitly in `ManagedSeed` (only via a `RawExtension`), but it's perhaps still a good idea to avoid it.

**Why is this needed**:
As discussed in https://github.com/gardener/gardener/pull/3414#discussion_r563806533, it is inconsistent and confusing that `seedConfig` / `seedTemplate` of a ManagedSeed is validated, but the the `seedConfig` of the standalone `GardenletConfiguration` is not validated. Since this is potentially a breaking change, we agreed to do it in a separate PR.

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Changes from `Seed` to `SeedTemplate` in `GardenletConfiguration`.

**Which issue(s) this PR fixes**:
Fixes #3469

**Special notes for your reviewer**:

**Release note**:
```other operator
NONE
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Comment by stoyanr** [comment]
There is a bug I just found that needs to be fixed, please don't merge. I converted the PR to draft, will reopen it back when I am done.

**Comment by stoyanr** [comment]
The issue is fixed now, the PR can be merged if there are no more comments.

File: pkg/gardenlet/apis/config/v1alpha1/types.go
Comment (timebertt) [inline pkg/gardenlet/apis/config/v1alpha1/types.go:350]: WDYT about removing the `SeedConfig` type in favor of the new `SeedTemplate`?

We might even consider renaming the `seedConfig` field to `seedTemplate` to make it consistent with the `ManagedSeed`.
Yes, I know, this is a breaking change. But the gardenlet config API is only in an `alpha` version, which means we are able to introduce breaking changes by API conventions.
Though, this is probably too much effort for such a cosmetic change and is not worth it.
WDYT?

File: pkg/gardenlet/controller/seed/seed.go
Comment (timebertt) [inline pkg/gardenlet/controller/seed/seed.go]: Nit: Can you move this up, so it's outside of `CreateOrUpdate`?
It doesn't depend on the object fetched from the API server.

**Review by timebertt — COMMENTED** [review]
Thanks a lot for cleaning this up! I think, it's much cleaner now.
Only one question and one minor nit from my side, rest looks good.

File: pkg/gardenlet/apis/config/v1alpha1/types.go
Comment (stoyanr) [reply pkg/gardenlet/apis/config/v1alpha1/types.go:350]: Yes, I think it's not worth it. As it is an existing YAML gardenlet configuration is basically compatible, even if a different type is being used. To introduce this we should either create a new version (`v1alpha2`) and take care of conversions, or we should be prepared to break all existing gardenlet configurations ... I don't think we want to spend the time and effort to fix them all.

File: pkg/gardenlet/controller/seed/seed.go
Comment (stoyanr) [reply pkg/gardenlet/controller/seed/seed.go]: Done

**Comment by stoyanr** [comment]
Rebased on top of latest master, fixed managed seed admission, and added a new commit with code review changes.
/squash

**Review by rfranzke — APPROVED** [review]

File: pkg/gardenlet/apis/config/v1alpha1/types.go
Comment (timebertt) [reply pkg/gardenlet/apis/config/v1alpha1/types.go:350]: Yes, agreed.

What about the first question: should we remove the `SeedConfig` type and change the `seedConfig` field to be of type `SeedTemplate` (no indirection)?

Comment (stoyanr) [reply pkg/gardenlet/apis/config/v1alpha1/types.go:350]: I don't like `seedConfig` of type `SeedTemplate`, and I don't think the indirection hurts in any way. I would vote for no more changes.

Could we merge this PR already?

Comment (timebertt) [reply pkg/gardenlet/apis/config/v1alpha1/types.go:350]: Ok, fine for me.

**Review by timebertt — APPROVED** [review]

**Pull request merged**
